src/include/parsing/parsing_table.h

```diff
@@ -0,0 +1,71 @@
+#ifndef JUCC_PARSING_PARSING_TABLE_H
+#define JUCC_PARSING_PARSING_TABLE_H
+
+#include <string>
+#include <unordered_map>
+
+#include "grammar/grammar.h"
+#include "utils/first_follow.h"
+
+namespace jucc {
+
+namespace parsing {
+
+class ParsingTable {
+  using Table = std::unordered_map<std::string, std::unordered_map<std::string, std::string> >;
+
+  /**
+   * stores the parsing table, which is calculated from the productions in the grammar
+   * and it's first and follow set
+   */
+  Table table_;
+
+  /**
+   * Stores firsts of the productions
+   */
+  utils::SymbolsMap firsts_;
+
+  /**
+   * Stores the follows of the productions
+   */
+  utils::SymbolsMap follows_;
+
+  /**
+   * Stores the productions of the grammar.
+   */
+  grammar::Productions productions_;
+
```

Comment:
we also need to keep track of terminals, non terminals and start symbol parsed in grammar parser.

Reply:
First and follow maps have the non-terminals.
From the productions, one can compute all the terminals.

Reply:
why recompute when that is already computed once?